apps/ex_wire/test/ex_wire/struct/bucket_test.exs

```diff
@@ -0,0 +1,69 @@
+defmodule ExWire.Struct.BucketTest do
+  use ExUnit.Case, async: true
+
+  doctest ExWire.Struct.Bucket
+
+  alias ExWire.Struct.Bucket
+  alias ExWire.KademliaConfig
+
+  describe "add_node/3" do
+    setup do
+      bucket = ExWire.Struct.Bucket.new(time: :test)
+
+      [bucket: bucket]
+    end
+
+    test "inserts new node to bucket", %{bucket: bucket} do
+      node =
+        ExWire.Struct.Peer.new(
+          "13.84.180.140",
+          30303,
+          "30b7ab30a01c124a6cceca36863ece12c4f5fa68e3ba9b0b51407ccc002eeed3b3102d20a88f1c1d3c3154e2449317b8ef95090e77b312d5cc39354f86d5d606"
+        )
+
+      {:insert_node, ^node, bucket} = bucket |> Bucket.add_node(node)
+
+      assert Bucket.member?(bucket, node)
+    end
+
+    test "reinsert node to bucket", %{bucket: bucket} do
+      node =
+        ExWire.Struct.Peer.new(
+          "13.84.180.140",
+          30303,
+          "30b7ab30a01c124a6cceca36863ece12c4f5fa68e3ba9b0b51407ccc002eeed3b3102d20a88f1c1d3c3154e2449317b8ef95090e77b312d5cc39354f86d5d606"
+        )
+
+      {:reinsert_node, ^node, bucket} =
+        bucket
+        |> Bucket.insert_node(node)
+        |> Bucket.add_node(node)
```

Comment:
yep, naming matters :)

`insert_node` followed by the `add_node` looks slightly confusing to me, imagine instead:

```elixir
      {:reinsert_node, ^node, bucket} =
        bucket
        |> Bucket.insert_node(node)
        |> Bucket.refresh_node(node)
```

Reply:
Actually `insert_node` should be private.
More correct version:
```
 bucket
 |> Bucket.add_node(node)
 |> Bucket.add_node(node)
```